web/src/routes/admin/library-management/+page.svelte

```diff
@@ -243,6 +243,19 @@
     updateLibraryIndex = selectedLibraryIndex;
   };
 
+  const onToggleReadOnlyClicked = async () => {
+    const library = libraries[selectedLibraryIndex];
+    const prevValue = library.isReadOnly == null ? library.type === LibraryType.External : library.isReadOnly;
```

Comment:
isReadOnly can't be null. I think you just want if(!external) { return } otherwise flip the value.

Reply:
ah, yep